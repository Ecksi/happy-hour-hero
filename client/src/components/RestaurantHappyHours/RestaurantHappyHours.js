import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './RestaurantHappyHours.css';

export class RestaurantHappyHours extends Component {
  getDrinkSpecials = () => {
    let drinkSpecials = [];
  
    this.props.todaysHappyHours.forEach(happyHour => {
      if (happyHour.combined_times === this.props.times) {
        this.props.hourlyDrinkSpecials.forEach(drinkSpecial => {
          if (drinkSpecial.id === happyHour.drink_specials_id && !drinkSpecials.includes(drinkSpecial.name)) {
            drinkSpecials.push(drinkSpecial.name);
          }
        });
      }
    });

    return drinkSpecials;
  }

  getFoodSpecials = () => {
    let foodSpecials = [];
  
    this.props.todaysHappyHours.forEach(happyHour => {
      if (happyHour.combined_times === this.props.times) {
        this.props.hourlyFoodSpecials.forEach(foodSpecial => {
          if (foodSpecial.id === happyHour.food_specials_id && !foodSpecials.includes(foodSpecial.name)) {
            foodSpecials.push(foodSpecial.name);
          }
        });
      }
    });

    return foodSpecials;
  }

  render() {
    const drinkSpecials = this.getDrinkSpecials();
    const foodSpecials = this.getFoodSpecials();

    return (
      <section className="restaurantHappyHoursContainer">
        <h3>{ this.props.times }</h3>
        <section className="restaurantSpecials">
          <article className="restaurantDrinkSpecials special">
            <h4>Drinks</h4>
            { drinkSpecials.length ? drinkSpecials.map((special, index) => <p key={index}>{special}</p>) : <p>No drink specials</p> }
          </article>
          <article className="restaurantFoodSpecials special">
            <h4>Food</h4>
            { foodSpecials.length ? foodSpecials.map((special, index) => <p key={index}>{special}</p>) : <p>No food specials</p> }
          </article>
        </section>
      </section>
    );
  }
}

RestaurantHappyHours.propTypes = {
  getRestaurant: PropTypes.func,
  getTodaysHappyHours: PropTypes.func,
  cleanHappyHourTimes: PropTypes.func,
  times: PropTypes.string,
  todaysHappyHours: PropTypes.array,
  hourlyDrinkSpecials: PropTypes.array,
  hourlyFoodSpecials: PropTypes.array,
};

export const mapStateToProps = (state) => ({
  location: state.location,
  drinkSpecials: state.drinkSpecials,
  foodSpecials: state.foodSpecials
});

export default connect(mapStateToProps)(RestaurantHappyHours);
