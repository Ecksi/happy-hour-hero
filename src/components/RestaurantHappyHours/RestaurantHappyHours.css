.restaurantHappyHoursContainer {
  width: 80%;
  padding: 50px 50px 0 50px;
  margin: 0 auto; }

.restaurantHappyHoursContainer h1 {
  color: #666;
  font-size: 2rem; }

.restaurantHappyHoursContainer h2 {
  color: #fdb82a;
  width: auto;
  font-size: 1.6rem;
  display: inline-block;
  padding: 20px;
  box-shadow: 1px 1px 1px 1px #ccc;
  border-radius: 5px;
  margin-top: 0; }

.restaurantHappyHoursContainer h3 {
  font-size: 1.6rem;
  color: #999;
  margin-top: 15px; }

.restaurantHappyHoursContainer p {
  color: #666;
  font-size: 1.1rem; }

.restaurantSpecials {
  display: flex;
  justify-content: space-between; }

.special {
  flex-direction: column;
  width: 46%;
  display: flex; }
