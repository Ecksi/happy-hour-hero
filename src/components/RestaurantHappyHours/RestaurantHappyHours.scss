.restaurantHappyHoursContainer {
  width: 80%;
  padding: 0px 50px 50px 50px;
  margin: 0 auto;
}

.restaurantContainer h2 {
  color: #666;
  width: 80%;
  font-size: 2rem;
  margin: 50px auto 50px auto;
}

.restaurantHappyHoursContainer h3 {
  color: #fdb82a;
  width: auto;
  font-size: 1.6rem;
  display: inline-block;
  border-radius: 5px;
  margin-top: 0;
}

.restaurantHappyHoursContainer h4 {
  font-size: 1.6rem;
  color: #999;
  margin-top: 15px;
}

.restaurantHappyHoursContainer p {
  color: #666;
  font-size: 1.1rem;
  margin: 10px 0 10px 0;
}

.restaurantSpecials {
  display: flex;
  justify-content: space-between;
}

.special {
  flex-direction: column;
  width: 46%;
  display: flex;
}