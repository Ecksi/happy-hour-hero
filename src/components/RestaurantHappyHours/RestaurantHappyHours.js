import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './RestaurantHappyHours.css';

class RestaurantHappyHours extends Component {
  getDrinkSpecials = (happyHours) => {
    let drinkSpecials = [];

    happyHours.forEach(happyHour => {
      if (happyHour.combined_times === this.props.times) {
        this.props.drinkSpecials.forEach(drinkSpecial => {
          if (drinkSpecial.id === happyHour.drink_specials_id && !drinkSpecials.includes(drinkSpecial.name)) {
            drinkSpecials.push(drinkSpecial.name);
          }
        });
      }
    });

    return drinkSpecials;
  }

  getFoodSpecials = (happyHours) => {
    let foodSpecials = [];

    happyHours.forEach(happyHour => {
      if (happyHour.combined_times === this.props.times) {
        this.props.foodSpecials.forEach(foodSpecial => {
          if (foodSpecial.id === happyHour.food_specials_id && !foodSpecials.includes(foodSpecial.name)) {
            foodSpecials.push(foodSpecial.name);
          }
        });
      }
    });

    return foodSpecials;
  }

  render() {
    const { times } = this.props;
    const happyHours = this.props.getTodaysHappyHours();
    const drinkSpecials = this.getDrinkSpecials(happyHours);
    const foodSpecials = this.getFoodSpecials(happyHours);
    
    return (
      <section className="restaurantHappyHoursContainer">
        <h3>{ this.props.times }</h3>
        <section className="restaurantSpecials">
          <article className="restaurantDrinkSpecials special">
            <h4>Drinks</h4>
            { drinkSpecials.length ? drinkSpecials.map((special, index) => <p key={index}>{special}</p>) : <p>No drink specials</p> }
          </article>
          <article className="restaurantFoodSpecials special">
            <h4>Food</h4>
            { foodSpecials.length ? foodSpecials.map((special, index) => <p key={index}>{special}</p>) : <p>No food specials</p> }
          </article>
        </section>
      </section>
    );
  }
}

RestaurantHappyHours.propTypes = {
  getRestaurant: PropTypes.func,
  getTodaysHappyHours: PropTypes.func,
  cleanHappyHourTimes: PropTypes.func,
};

export const mapStateToProps = (state) => ({
  location: state.location,
  drinkSpecials: state.drinkSpecials,
  foodSpecials: state.foodSpecials
});

export default connect(mapStateToProps)(RestaurantHappyHours);
