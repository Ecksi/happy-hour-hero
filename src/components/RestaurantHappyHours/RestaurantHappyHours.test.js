import React from 'react';
import { shallow } from 'enzyme';
import { RestaurantHappyHours } from './RestaurantHappyHours';

describe('RestaurantHappyHours', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      todaysHappyHours: [
        {
          combined_times:'4:00PM-8:00PM',
          day: 'Monday',
          drink_specials_id: 5,
          end_time:'2000',
          food_specials_id: null,
          id: 1,
          restaurant_id: 1,
          start_time:'1600',
        }
      ],
      hourlyDrinkSpecials: [
        {
          best_deal: true,
          id: 5,
          name: "2-for-1 drinks",
        }
      ],
      hourlyFoodSpecials: [],
      times: '4:00PM-8:00PM'
    };
    wrapper = shallow(<RestaurantHappyHours {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('getDrinkSpecials', () => {
  
    beforeEach(() => {
      wrapper = shallow(<RestaurantHappyHours {...mockProps} />);
    });
  
    it('should return an array of drink specials', () => {
      const result = wrapper.instance().getDrinkSpecials();

      expect(result).toEqual(["2-for-1 drinks"]);
    });
  });
});