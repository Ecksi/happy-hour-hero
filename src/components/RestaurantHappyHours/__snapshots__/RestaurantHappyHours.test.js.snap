// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantHappyHours should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantHappyHours
    hourlyDrinkSpecials={
        Array [
            Object {
              "best_deal": true,
              "id": 5,
              "name": "2-for-1 drinks",
            },
          ]
    }
    hourlyFoodSpecials={Array []}
    times="4:00PM-8:00PM"
    todaysHappyHours={
        Array [
            Object {
              "combined_times": "4:00PM-8:00PM",
              "day": "Monday",
              "drink_specials_id": 5,
              "end_time": "2000",
              "food_specials_id": null,
              "id": 1,
              "restaurant_id": 1,
              "start_time": "1600",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          4:00PM-8:00PM
</h3>,
        <section
          className="restaurantSpecials"
>
          <article
                    className="restaurantDrinkSpecials special"
          >
                    <h4>
                              Drinks
                    </h4>
                    <p>
                              2-for-1 drinks
                    </p>
          </article>
          <article
                    className="restaurantFoodSpecials special"
          >
                    <h4>
                              Food
                    </h4>
                    <p>
                              No food specials
                    </p>
          </article>
</section>,
      ],
      "className": "restaurantHappyHoursContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "4:00PM-8:00PM",
        },
        "ref": null,
        "rendered": "4:00PM-8:00PM",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="restaurantDrinkSpecials special"
>
              <h4>
                            Drinks
              </h4>
              <p>
                            2-for-1 drinks
              </p>
</article>,
            <article
              className="restaurantFoodSpecials special"
>
              <h4>
                            Food
              </h4>
              <p>
                            No food specials
              </p>
</article>,
          ],
          "className": "restaurantSpecials",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Drinks
</h4>,
                Array [
                  <p>
                    2-for-1 drinks
</p>,
                ],
              ],
              "className": "restaurantDrinkSpecials special",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Drinks",
                },
                "ref": null,
                "rendered": "Drinks",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "2-for-1 drinks",
                },
                "ref": null,
                "rendered": "2-for-1 drinks",
                "type": "p",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Food
</h4>,
                <p>
                  No food specials
</p>,
              ],
              "className": "restaurantFoodSpecials special",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Food",
                },
                "ref": null,
                "rendered": "Food",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No food specials",
                },
                "ref": null,
                "rendered": "No food specials",
                "type": "p",
              },
            ],
            "type": "article",
          },
        ],
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            4:00PM-8:00PM
</h3>,
          <section
            className="restaurantSpecials"
>
            <article
                        className="restaurantDrinkSpecials special"
            >
                        <h4>
                                    Drinks
                        </h4>
                        <p>
                                    2-for-1 drinks
                        </p>
            </article>
            <article
                        className="restaurantFoodSpecials special"
            >
                        <h4>
                                    Food
                        </h4>
                        <p>
                                    No food specials
                        </p>
            </article>
</section>,
        ],
        "className": "restaurantHappyHoursContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "4:00PM-8:00PM",
          },
          "ref": null,
          "rendered": "4:00PM-8:00PM",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article
                className="restaurantDrinkSpecials special"
>
                <h4>
                                Drinks
                </h4>
                <p>
                                2-for-1 drinks
                </p>
</article>,
              <article
                className="restaurantFoodSpecials special"
>
                <h4>
                                Food
                </h4>
                <p>
                                No food specials
                </p>
</article>,
            ],
            "className": "restaurantSpecials",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Drinks
</h4>,
                  Array [
                    <p>
                      2-for-1 drinks
</p>,
                  ],
                ],
                "className": "restaurantDrinkSpecials special",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Drinks",
                  },
                  "ref": null,
                  "rendered": "Drinks",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "2-for-1 drinks",
                  },
                  "ref": null,
                  "rendered": "2-for-1 drinks",
                  "type": "p",
                },
              ],
              "type": "article",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Food
</h4>,
                  <p>
                    No food specials
</p>,
                ],
                "className": "restaurantFoodSpecials special",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Food",
                  },
                  "ref": null,
                  "rendered": "Food",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No food specials",
                  },
                  "ref": null,
                  "rendered": "No food specials",
                  "type": "p",
                },
              ],
              "type": "article",
            },
          ],
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
